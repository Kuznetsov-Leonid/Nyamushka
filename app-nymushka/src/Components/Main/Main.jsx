/**
 * create by Kuznetsov Leonid
 */

import React, { Component } from "react";
import { Item } from "../Card/CardItem";
import { DATA } from '../../data/data';

class Main extends Component {
    constructor(props) {
        super(props);
        this.state = DATA;
        this.toggleStatus = this.toggleStatus.bind(this);
        this.getItemIndexById = this.getItemIndexById.bind(this);
    }

    toggleStatus(id) {
        let index = this.getItemIndexById(id);
        let newItems = this.state.items;
        if (newItems[index].status !== "disabled"){
            newItems[index].status = (newItems[index].status === "notSelected") ? "selected" : "notSelected";
            this.setState({ items: newItems });
        }
    }

    getItemIndexById(id) {
        let res = -1;
        this.state.items.map((item, index) => {
            if (item.id === id) res = index;
        });
        return res;
    }

    render() {
        return (
            <div class="container px-2 px-xl-0">
                <div class="row mx-0">
                    <div class="col-xl-4 d-flex justify-content-center mb-4 mb-xl-0">
                        <Item item={this.state.items[0]} toggleStatus={this.toggleStatus} />
                    </div>
                    <div class="col-xl-4 col-12 col-md-6 d-flex justify-content-center mb-4 mb-md-0">
                        <Item item={this.state.items[1]} toggleStatus={this.toggleStatus} />
                    </div>
                    <div class="col-xl-4 col-12 col-md-6 d-flex justify-content-center mb-4 mb-md-0">
                        <Item item={this.state.items[2]} toggleStatus={this.toggleStatus} />
                    </div>
                </div>
            </div>
        );
    }
}

export default Main;