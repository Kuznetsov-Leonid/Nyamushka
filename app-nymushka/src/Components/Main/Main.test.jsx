import { render, screen } from '@testing-library/react';
import Main from './Main';


describe('Main components', () => {
    //Простой рендер компонентов
    it('Main render', () => {
        render(<Main />);
        expect(screen.queryByRole('Main')).toBeNull();
    });
});

describe('Mainw snapshot', () => {
    it('Main card snapshot', () => {
        const MainSnap = render(<Main />);
        expect(MainSnap).toMatchSnapshot();
    });
});
