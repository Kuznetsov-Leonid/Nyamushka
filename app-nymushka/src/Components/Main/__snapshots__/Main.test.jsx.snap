// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainw snapshot Main card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container px-2 px-xl-0"
      >
        <div
          class="row mx-0"
        >
          <div
            class="col-xl-4 d-flex justify-content-center mb-4 mb-xl-0"
          >
            <div
              class="Item-wrapper notSelected"
            >
              <div
                class="Item"
              >
                <div
                  class="Item__inner-wrapper"
                />
                <div
                  class="Item__title"
                >
                  Сказочное заморское яство
                </div>
                <h3
                  class="Item__name"
                >
                  Нямушка
                </h3>
                <div
                  class="Item__stuffing"
                >
                  с фуа-гра
                </div>
                <div
                  class="Item__portionCount"
                >
                  <b>
                    10
                  </b>
                   порций
                </div>
                <div
                  class="Item__gift"
                >
                  мышь в подарок
                </div>
                <div
                  class="Item__nyamushka"
                />
                <div
                  class="Item__weight"
                >
                  <div
                    class="Item__weight_value"
                  >
                    0,5
                  </div>
                  <div
                    class="Item__weight_kg"
                  >
                    кг
                  </div>
                </div>
              </div>
              <div
                class="Item__label Item__label_buy"
              >
                Чего сидишь? Порадуй котэ, 
                <a
                  class="Item__label_link"
                  href="#"
                >
                  купи
                </a>
                <pointblu>
                  .
                </pointblu>
              </div>
              <div
                class="Item__label Item__label_sadness"
              >
                Печалька, с фуа-гра закончился.
              </div>
              <div
                class="Item__label Item__label_selected"
              >
                Печень утки разварная с артишоками.
              </div>
            </div>
          </div>
          <div
            class="col-xl-4 col-12 col-md-6 d-flex justify-content-center mb-4 mb-md-0"
          >
            <div
              class="Item-wrapper selected"
            >
              <div
                class="Item"
              >
                <div
                  class="Item__inner-wrapper"
                />
                <div
                  class="Item__title"
                >
                  Сказочное заморское яство
                </div>
                <h3
                  class="Item__name"
                >
                  Нямушка
                </h3>
                <div
                  class="Item__stuffing"
                >
                  с рыбой
                </div>
                <div
                  class="Item__portionCount"
                >
                  <b>
                    40
                  </b>
                   порций
                </div>
                <div
                  class="Item__gift"
                >
                  2 мыши в подарок
                </div>
                <div
                  class="Item__nyamushka"
                />
                <div
                  class="Item__weight"
                >
                  <div
                    class="Item__weight_value"
                  >
                    2
                  </div>
                  <div
                    class="Item__weight_kg"
                  >
                    кг
                  </div>
                </div>
              </div>
              <div
                class="Item__label Item__label_buy"
              >
                Чего сидишь? Порадуй котэ, 
                <a
                  class="Item__label_link"
                  href="#"
                >
                  купи
                </a>
                <pointblu>
                  .
                </pointblu>
              </div>
              <div
                class="Item__label Item__label_sadness"
              >
                Печалька, с рыбой закончился.
              </div>
              <div
                class="Item__label Item__label_selected"
              >
                Головы щучьи с чесноком да свежайшая сёмгушка.
              </div>
            </div>
          </div>
          <div
            class="col-xl-4 col-12 col-md-6 d-flex justify-content-center mb-4 mb-md-0"
          >
            <div
              class="Item-wrapper disabled"
            >
              <div
                class="Item"
              >
                <div
                  class="Item__inner-wrapper"
                />
                <div
                  class="Item__title"
                >
                  Сказочное заморское яство
                </div>
                <h3
                  class="Item__name"
                >
                  Нямушка
                </h3>
                <div
                  class="Item__stuffing"
                >
                  с курой
                </div>
                <div
                  class="Item__portionCount"
                >
                  <b>
                    100
                  </b>
                   порций
                </div>
                <div
                  class="Item__gift"
                >
                  5 мышей в подарок
                </div>
                <div
                  class="Item__gift"
                >
                  заказчик доволен
                </div>
                <div
                  class="Item__nyamushka"
                />
                <div
                  class="Item__weight"
                >
                  <div
                    class="Item__weight_value"
                  >
                    5
                  </div>
                  <div
                    class="Item__weight_kg"
                  >
                    кг
                  </div>
                </div>
              </div>
              <div
                class="Item__label Item__label_buy"
              >
                Чего сидишь? Порадуй котэ, 
                <a
                  class="Item__label_link"
                  href="#"
                >
                  купи
                </a>
                <pointblu>
                  .
                </pointblu>
              </div>
              <div
                class="Item__label Item__label_sadness"
              >
                Печалька, с курой закончился.
              </div>
              <div
                class="Item__label Item__label_selected"
              >
                Филе из цыплят с трюфелями в бульоне.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container px-2 px-xl-0"
    >
      <div
        class="row mx-0"
      >
        <div
          class="col-xl-4 d-flex justify-content-center mb-4 mb-xl-0"
        >
          <div
            class="Item-wrapper notSelected"
          >
            <div
              class="Item"
            >
              <div
                class="Item__inner-wrapper"
              />
              <div
                class="Item__title"
              >
                Сказочное заморское яство
              </div>
              <h3
                class="Item__name"
              >
                Нямушка
              </h3>
              <div
                class="Item__stuffing"
              >
                с фуа-гра
              </div>
              <div
                class="Item__portionCount"
              >
                <b>
                  10
                </b>
                 порций
              </div>
              <div
                class="Item__gift"
              >
                мышь в подарок
              </div>
              <div
                class="Item__nyamushka"
              />
              <div
                class="Item__weight"
              >
                <div
                  class="Item__weight_value"
                >
                  0,5
                </div>
                <div
                  class="Item__weight_kg"
                >
                  кг
                </div>
              </div>
            </div>
            <div
              class="Item__label Item__label_buy"
            >
              Чего сидишь? Порадуй котэ, 
              <a
                class="Item__label_link"
                href="#"
              >
                купи
              </a>
              <pointblu>
                .
              </pointblu>
            </div>
            <div
              class="Item__label Item__label_sadness"
            >
              Печалька, с фуа-гра закончился.
            </div>
            <div
              class="Item__label Item__label_selected"
            >
              Печень утки разварная с артишоками.
            </div>
          </div>
        </div>
        <div
          class="col-xl-4 col-12 col-md-6 d-flex justify-content-center mb-4 mb-md-0"
        >
          <div
            class="Item-wrapper selected"
          >
            <div
              class="Item"
            >
              <div
                class="Item__inner-wrapper"
              />
              <div
                class="Item__title"
              >
                Сказочное заморское яство
              </div>
              <h3
                class="Item__name"
              >
                Нямушка
              </h3>
              <div
                class="Item__stuffing"
              >
                с рыбой
              </div>
              <div
                class="Item__portionCount"
              >
                <b>
                  40
                </b>
                 порций
              </div>
              <div
                class="Item__gift"
              >
                2 мыши в подарок
              </div>
              <div
                class="Item__nyamushka"
              />
              <div
                class="Item__weight"
              >
                <div
                  class="Item__weight_value"
                >
                  2
                </div>
                <div
                  class="Item__weight_kg"
                >
                  кг
                </div>
              </div>
            </div>
            <div
              class="Item__label Item__label_buy"
            >
              Чего сидишь? Порадуй котэ, 
              <a
                class="Item__label_link"
                href="#"
              >
                купи
              </a>
              <pointblu>
                .
              </pointblu>
            </div>
            <div
              class="Item__label Item__label_sadness"
            >
              Печалька, с рыбой закончился.
            </div>
            <div
              class="Item__label Item__label_selected"
            >
              Головы щучьи с чесноком да свежайшая сёмгушка.
            </div>
          </div>
        </div>
        <div
          class="col-xl-4 col-12 col-md-6 d-flex justify-content-center mb-4 mb-md-0"
        >
          <div
            class="Item-wrapper disabled"
          >
            <div
              class="Item"
            >
              <div
                class="Item__inner-wrapper"
              />
              <div
                class="Item__title"
              >
                Сказочное заморское яство
              </div>
              <h3
                class="Item__name"
              >
                Нямушка
              </h3>
              <div
                class="Item__stuffing"
              >
                с курой
              </div>
              <div
                class="Item__portionCount"
              >
                <b>
                  100
                </b>
                 порций
              </div>
              <div
                class="Item__gift"
              >
                5 мышей в подарок
              </div>
              <div
                class="Item__gift"
              >
                заказчик доволен
              </div>
              <div
                class="Item__nyamushka"
              />
              <div
                class="Item__weight"
              >
                <div
                  class="Item__weight_value"
                >
                  5
                </div>
                <div
                  class="Item__weight_kg"
                >
                  кг
                </div>
              </div>
            </div>
            <div
              class="Item__label Item__label_buy"
            >
              Чего сидишь? Порадуй котэ, 
              <a
                class="Item__label_link"
                href="#"
              >
                купи
              </a>
              <pointblu>
                .
              </pointblu>
            </div>
            <div
              class="Item__label Item__label_sadness"
            >
              Печалька, с курой закончился.
            </div>
            <div
              class="Item__label Item__label_selected"
            >
              Филе из цыплят с трюфелями в бульоне.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
